pkg/netpol/connlist/connlist.go

```diff
@@ -415,37 +416,70 @@ func convertEvalPeersToConnlistPeer(peers []eval.Peer) []Peer {
 	return res
 }
 
-// getConnectionsList returns connections list from PolicyEngine and ingressAnalyzer objects
-// if the exposure-analysis option is on, also computes and updates the exposure-analysis results
-func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine, ia *ingressanalyzer.IngressAnalyzer) ([]Peer2PeerConnection,
-	[]Peer, error) {
-	connsRes := make([]Peer2PeerConnection, 0)
-	if !pe.HasPodPeers() {
-		return connsRes, []Peer{}, nil
+// getPeersForConnsComputation returns two slices of src and dst peers and a union slice of all peers (disjoint);
+func (ca *ConnlistAnalyzer) getPeersForConnsComputation(pe *eval.PolicyEngine) (srcPeers, dstPeers, peers []Peer, err error) {
+	// get ip-block peers (src ip-block and dst ip-blocks and disjoint of both) extracted from policy rules
+	srcIpbList, dstIpbList, disjointOfAllIPs, err := pe.GetIPBlockPeersLists()
+	if err != nil {
+		ca.errors = append(ca.errors, newResourceEvaluationError(err))
+		return nil, nil, nil, err
 	}
+	// initiate results slices with IpBlock peers (peers are  converted []connlist.Peer list to be used in connlist pkg and returned)
+	srcPeers = convertEvalPeersToConnlistPeer(srcIpbList)
+	dstPeers = convertEvalPeersToConnlistPeer(dstIpbList)
+	peers = convertEvalPeersToConnlistPeer(disjointOfAllIPs)
 
-	// get workload peers and ip blocks
+	// get workload peers - peers from manifests
 	peerList, err := pe.GetPeersList()
```

Comment:
rename `GetPeersList` to `GetWorkloadPeersList`

Reply:
I am afraid this would be an API break? wouldn't be?

Reply:
how about having a new `GetWorkloadPeersList` func that returns only workload peers, and changing this function to return the union of workload peers and ip block peers, as it was before?

Reply:
done